Add Universe, SolarSystem, and Planet classes

rohanrk opened the pull request.

rohanrk committed:
Add Universe, SolarSystem, and Planet classes

rohanrk commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Solar System and Universe Model](https://trello.com/c/qOgLIHlV/59-solar-system-and-universe-model)

bayne-grubb commented on app/src/main/java/com/communistutopia/spacetrader/model/Universe.kt:
```diff
@@ -0,0 +1,78 @@
+    init {
+        this.planet = planet
+        this.name = name
+        this.tech = tech
```
In the meeting we discussed having tech level as a class belonging to Planet, not solar system. I would suggest removing this and your planet class.

rohanrk replied:
Demo requirements state that the solar system has to have a tech level. We could refactor this later. I'll add a comment

rohanrk replied:
I take it all back. I'll refactor

https://piazza.com/class/jqffgcxzzhz2v9?cid=236

bayne-grubb commented on app/src/main/java/com/communistutopia/spacetrader/model/Universe.kt:
```diff
@@ -0,0 +1,78 @@
+
+class SolarSystem(planet: Planet, name: String, tech: Int = 0, x: Int, y: Int) {
+
+    var planet: Planet
```
We also talked about having multiple planets per SS. Perhaps this could be refactored as a Collection of planets? Unsure about how that would affect your randomization logic

bayne-grubb commented on app/src/main/java/com/communistutopia/spacetrader/model/Universe.kt:
```diff
@@ -0,0 +1,78 @@
+        random = Random.Default
+    }
+
+    fun generateUniverse() {
```
This is really good and also will make our lives easier as far as creating the universe goes, nice job :)
One thing I don't understand is how you're generating the names? Maybe comment those two or three lines

bayne-grubb replied:
Also wait is this an infinite loop? Where is i decremented

haleylittmann commented on app/src/main/java/com/communistutopia/spacetrader/model/Universe.kt:
```diff
@@ -0,0 +1,78 @@
+    private val Y_SIZE: Int = 100
+    private val NUM_SYSTEMS: Int = 10
+
+    private val solarSystems: ArrayList<SolarSystem> = ArrayList()
```
I think Dr. Waters spoke in class about how we should program to abstractions, like List<SolarSystem>sys = new ArrayList<>() (in however Kotlin does it)

haleylittmann commented on app/src/main/java/com/communistutopia/spacetrader/model/Universe.kt:
```diff
@@ -0,0 +1,78 @@
+                var name: String = "System%d".format(i)
+                solarSystems.add(SolarSystem(planet = Planet(""), name = name, x = x, y = y))
+                universe[pair] = name
+            }
```
I'm assuming you'd decrement i in the if statement otherwise it wouldn't properly make enough systems right?

haleylittmann commented on app/src/main/java/com/communistutopia/spacetrader/model/Universe.kt:
```diff
@@ -0,0 +1,78 @@
+        random = Random.Default
+    }
+
+    fun generateUniverse() {
```
Also  please javadoc or kotlin doc, so we won't have to come back and do it, then it's also easier to see exactly what is happening in the method :)

haleylittmann commented on app/src/main/java/com/communistutopia/spacetrader/model/Universe.kt:
```diff
@@ -0,0 +1,78 @@
+    private val Y_SIZE: Int = 100
+    private val NUM_SYSTEMS: Int = 10
+
+    private val solarSystems: ArrayList<SolarSystem> = ArrayList()
```
Dr. Waters spoke in class about how we should program to abstractions, like List<Solar System> sys = new ArrayList<>() (in however Kotlin does it)

rohanrk replied:
This isn't going to work in Kotlin. For some bizarre reason, the Kotlin developers didn't put an add method in their generic List/Map interfaces so the compiler will get mad if I change this to a less specific structure.

I'll check with Waters tomorrow

rohanrk committed:
Fix infinite loop bug

rohanrk committed:
Fix infinite loop bug

rohanrk committed:
Universe.kt: Add kotlin doc comments and refactor

Changed Solar Systems collection to Set and created planets Set.
Increases efficiency of lookups and allows solar systems to have
a variable number of planets.

Switched tech level from solar system class to planet class. Now
each planet can have different tech levels

rohanrk committed:
Universe.kt: Add kotlin doc comments and refactor

Changed Solar Systems collection to Set and created planets Set.
Increases efficiency of lookups and allows solar systems to have
a variable number of planets.

Switched tech level from solar system class to planet class. Now
each planet can have different tech levels

rohanrk committed:
Universe.kt: Add kotlin doc comments and refactor

Changed Solar Systems collection to Set and created planets Set.
Increases efficiency of lookups and allows solar systems to have
a variable number of planets.

Switched tech level from solar system class to planet class. Now
each planet can have different tech levels

rohanrk committed:
Add distance func that takes in locations

rohanrk committed:
Add distance func that takes in locations

rohanrk committed:
Add distance func that takes in locations

haleylittmann committed:
Merge branch 'model' of https://github.com/rohanrk/Space-Trader into model

haleylittmann committed:
New variable max number of planets, each planet can have 1-3 planets, deleted planet class

haleylittmann committed:
Added max number of planets, each solar system can have 1-3 planets,

haleylittmann committed:
Deleted planet class to be able to merge with Bayne

haleylittmann committed:
Fixed logic so only solar systems were added to universe

DahliaWitt commented:
Bayne's branch encompasses this.
